VultisigApp/VultisigApp/Services/Sui/CetusAggregatorService.swift

```diff
@@ -0,0 +1,370 @@
+                amount: amount
+            )
+            
+            // Calculate price from the response
+            if routes.routes.count > 0 {
+                let amountOut = Double(routes.amount_out) ?? 0
+                let amountIn = Double(routes.amount_in) ?? 0
+                
+                // Since we're swapping to USDC (6 decimals), we need to adjust
+                let usdcDecimals = 6
+                let tokenDecimals = 9 // Default for SUI tokens, should be fetched from metadata
+                
+                let usdcAmount = amountOut / pow(10, Double(usdcDecimals))
+                let tokenAmount = amountIn / pow(10, Double(tokenDecimals))
+                
+                // Price = USDC amount / Token amount
+                let price = tokenAmount > 0 ? usdcAmount / tokenAmount : 0
+                
+                return price > 0 ? price : 0.0
+            }
+            
+            // If no direct route to USDC, try via SUI
+            let suiAddress = "0x2::sui::SUI"
+            
+            // First get token -> SUI rate
+            let tokenToSuiRoutes = try await findRoutes(
+                fromToken: contractAddress,
+                toToken: suiAddress,
+                amount: amount
+            )
+            
+            if tokenToSuiRoutes.routes.count > 0 {
+                let suiAmountOut = tokenToSuiRoutes.amount_out
+                
+                // Then get SUI -> USDC rate
+                let suiToUsdcRoutes = try await findRoutes(
+                    fromToken: suiAddress,
+                    toToken: usdcAddress,
+                    amount: suiAmountOut
+                )
+                
+                if suiToUsdcRoutes.routes.count > 0 {
+                    let usdcAmountOut = Double(suiToUsdcRoutes.amount_out) ?? 0
+                    let amountIn = Double(amount) ?? 0
+                    
+                    // Calculate final price
+                    let usdcDecimals = 6
+                    let tokenDecimals = 9
+                    
+                    let usdcAmount = usdcAmountOut / pow(10, Double(usdcDecimals))
+                    let tokenAmount = amountIn / pow(10, Double(tokenDecimals))
+                    
+                    let price = tokenAmount > 0 ? usdcAmount / tokenAmount : 0
+                    
+                    return price > 0 ? price : 0.0
+                }
+            }
+            
+            return 0.0
+            
+        } catch {
+            return 0.0
+        }
+    }
+    
+    /// Get token price with proper decimal handling
+    /// - Parameters:
+    ///   - contractAddress: Token contract address
+    ///   - decimals: Token decimals
+    /// - Returns: Price in USD
+    func getTokenUSDValue(contractAddress: String, decimals: Int) async -> Double {
+        do {
+            let usdcAddress = "0xdba34672e30cb065b1f93e3ab55318768fd6fef66c15942c9f7cb846e2f900e7::usdc::USDC"
```

Comment:
this address has been used multiple place , it will be better if we define it somewhere